src/qt/walletcontroller.cpp

```diff
@@ -29,15 +33,17 @@ WalletController::WalletController(interfaces::Node& node, const PlatformStyle*
         getOrCreateWallet(std::move(wallet));
     }
 
-    m_activity_thread.start();
+    m_activity_worker->moveToThread(m_activity_thread);
+    m_activity_thread->start();
 }
 
 // Not using the default destructor because not all member types definitions are
 // available in the header, just forward declared.
 WalletController::~WalletController()
 {
-    m_activity_thread.quit();
-    m_activity_thread.wait();
+    m_activity_thread->quit();
+    m_activity_thread->wait();
+    delete m_activity_worker;
```

Comment:
Could naked `delete` be avoided in the new code? There are some techniques to achieve this.
E.g., `m_activity_worker` could have a parent `QObject`, like `m_activity_thread` has.

Reply:
No object could be `m_activity_worker` parent because it's the only object on its thread. Could connect `WalletController::destroyed` to its `deleteLater` or could use some smart pointer but honestly I think this is better.

Reply:
You could use `QThread::finished` signal. See: https://github.com/bitcoin/bitcoin/pull/16261/files#r303227866